test/kennel/parts_writer_test.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+require_relative "../test_helper"
+require "find"
+
+SingleCov.covered!
+
+describe Kennel::PartsWriter do
+  def write(file, content)
+    folder = File.dirname(file)
+    FileUtils.mkdir_p folder unless File.exist?(folder)
+    File.write file, content
+  end
+
+  let(:project_filter) { nil }
+  let(:tracking_id_filter) { nil }
+
+  let(:filter) do
+    filter = "some filter".dup
+    filter.stubs(:project_filter).returns(project_filter)
+    filter.stubs(:tracking_id_filter).returns(tracking_id_filter)
+    filter
+  end
+
+  capture_all
+
+  def make_project(kennel_id, monitor_kennel_ids)
+    Kennel::Models::Project.new(
+      team: Kennel::Models::Team.new(
+        kennel_id: "team-id",
+        mention: "@slack-whatever"
+      ),
+      name: kennel_id,
+      kennel_id: kennel_id,
+      parts: -> {
+        monitor_kennel_ids.map do |id|
+          Kennel::Models::Monitor.new(
+            self,
+            type: "query alert",
+            kennel_id: id,
+            query: "avg(last_5m) > 123",
+            critical: 123
+          )
+        end
+      }
+    )
+  end
+
+  in_temp_dir
+
+  it "saves formatted json" do
```

Comment:
wrap everything in `describe "#write"` to make adding more methods/tests not require a big reformat

Reply:
843c5ccfa1fb27302d2de788a7288ffbd5c273a2